ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -811,13 +811,13 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = electric $ do
     computeLoEMaxExtra (LoEEnabled loeFrag) newBlockFrag =
         -- Both fragments are on the same chain
         if loeSuffixLength == 0 || rollback == 0 then
-          if rollback > k + loeSuffixLength
-            then Nothing
-            else Just $ LoEEnabled $ k + loeSuffixLength - rollback
+          if rollback < k + loeSuffixLength
+            then Just $ LoEEnabled $ k + loeSuffixLength - rollback
+            else Nothing
         else
-          if rollback > k
-            then Nothing
-            else Just $ LoEEnabled $ k - rollback
+          if rollback < k
+            then Just $ LoEEnabled $ k - rollback
+            else Nothing
```

Comment:
This refactoring is changing the behavior of the code. In particular, it is returning `Nothing` where before it would return `Just 0`.

Reply:
Damn, you're right, what the hell.